mage/template.go

```diff
@@ -93,6 +93,60 @@ Options:
 		return
 	}
 	  
+	// Terminals with color support:
+	// 	"TERM=xterm"
+	// 	"TERM=xterm-vt220"
+	// 	"TERM=xterm-256color"
+	// 	"TERM=screen-256color"
+	// 	"TERM=tmux-256color"
+	// 	"TERM=rxvt-unicode-256color"
+	// Don't support color:
+	// 	"TERM=cygwin"
+	var specialColorTerms = map[string]bool{
+		"screen-256color":       true,
+		"tmux-256color":         true,
+		"rxvt-unicode-256color": true,
+	}
+
+	// terminalSupportsColor checks if the current console supports color output
+	//
+	// Supported:
+	// 	linux, mac, or windows's ConEmu, Cmder, putty, git-bash.exe
+	// Not supported:
+	// 	windows cmd.exe, powerShell.exe
+	terminalSupportsColor := func() bool {
+		envTerm := os.Getenv("TERM")
+		if strings.Contains(envTerm, "xterm") {
```

Comment:
This kinda skeeves me out.  Are we sure there are no existing terminals that include the substring xterm that don't support color? (I ask because I have no clue)

Reply:
Yeah, the best I could find was that there's "xterm-mono". I've added a condition to disable terminal support of this is detected.

See https://github.com/rust-lang/cargo/issues/7646

Reply:
Sorry, I missed this was a question for me. Yes, let's do that.

Reply:
Cool - done, see the update.

I've used a map rather than a list for performance, so that I don't need to iterate over it and just use the terminal name as a key.

Reply:
"It works on my machine" including all tests, but having some trouble with tests in Travis CI. Trying to find out what is the TERM set to...

Reply:
Looking good now. Ready for a merge & release?